core/src/main/java/org/mskcc/cbio/portal/dao/DaoCancerStudy.java

```diff
@@ -330,7 +327,10 @@ public static void addCancerStudy(CancerStudy cancerStudy, boolean overwrite) th
                 pstmt.setString(8, StringUtils.join(groups, ";"));
             }
             pstmt.setString(9, cancerStudy.getShortName());
-
+          	//status is UNAVAILABLE until other data is loaded for this study. Once all is loaded, the 
+            //data loading process can set this to AVAILABLE:
+            //TODO - use this field in parts of the system that build up the list of studies to display in home page:
+            pstmt.setInt(10, Status.UNAVAILABLE.ordinal());
```

Comment:
Yes please.  I agree on the need to do this, in fact we attempt to do this, but it has to integrate propertly with the primitive cache/hashmaps that are populated in memory and it was here that the code became problematic.  I would prefer to implement a complete solution in the PR request and not spread out across multiple ones.